irohad/synchronizer/impl/synchronizer_impl.cpp

```diff
@@ -63,14 +70,25 @@ namespace iroha {
     void SynchronizerImpl::process_commit(network::Commit commit_message) {
       log_->info("processing commit");
 
+      auto top_block_height = block_query_factory_->createBlockQuery() |
```

Comment:
This will result in `uint32_t()` (which is 0) being assigned to `top_block_height` if `createBlockQuery` returns `boost::none`. This should be either refactored with two different statements (one for creating block query and checking it, and another for getting top block height) or returning optional from lambda with getTopBlockHeight().